theseus/optimizer/nonlinear/nonlinear_optimizer.py

```diff
@@ -297,6 +302,7 @@ def _optimize_impl(
         track_err_history: bool = False,
         verbose: bool = False,
         backward_mode: BackwardMode = BackwardMode.FULL,
+        grouped_retract: bool = True,
```

Comment:
Maybe `batched_retract` since our terminology is `batched_variable` and `batched_cost_function`? Or should we rename `batched_xxx` back to `grouped_xxx`?